### DISCUSSION TITLE
Admin Access Error

### DISCUSSION THREAD

**Discussion by reluss** [balzack/databag#100] [Q&A] [answered]
Hello,
I'm having trouble accessing the admin panel for the first time.
I tried to set up the project on my Raspberry Pi Zero as described [here](https://github.com/balzack/databag/blob/main/doc/pizero.md). So far I'm running the Raspi on my local network only. And accessing the databag dashboard via web browser on my computer seems to work. But when I try to login as admin on the admin login panel it says "Please confirm your password.".
I don't remember what the input field showed the first time I visited that page. I assumed it wanted a new password from me and so I entered a pretty long password with 64 characters. Maybe some of those characters or symbols weren't supported or something, I don't know.
I checked the current password within the databag.db and even changed it there via a sqlite command to something easy with letters only but the problem still persists.
Is there a way to get it running or troubleshoot the issue and if so, how can I do it?

Thank you very much for that promising project and any help on my issue!

**Comment by balzack** [comment]
I am not sure I am understanding correctly, but it seems you want to login with a user account with the handle 'admin'?

Access to the admin dashboard is granted with the admin token, but there is no 'admin' user account unless you create one. From the admin dashboard you can generate a 'create account link'. When you follow that link you can choose a handle and password for the new account. You can choose 'admin' if you like.

Please let me know if I misunderstand the problem. Thanks for your interest in the project!

**Reply by reluss** [reply]
I can't get into the admin dashboard in the first place. Whenever I enter the password under DOMAIN/#/admin I get that mentioned error "Please confirm your password.".

**Comment by balzack** [comment]
When you refer to the admin dashboard not working, is this the mobile app, since it does work with the browser? Sorry I am having trouble understanding the difference in when it works and when it doesn't.

**Comment by reluss** [comment]
No, it did not work at all yet, not in the browser and not on mobile (app).
I installed the entire system and now I want to get into the admin dashboard so that I can generate the invite links. But what password do I have to use to get into the admin dashboard?

I also tried to get the password via:
`sqlite3 /var/lib/databag/databag.db`
`sqlite> select str_value from configs where config_id='token';`

I also changed the password there to an easy one without any symbols, but that didn't help either.
I still can't get into the admin dashboard.

I'm sorry if I give you a hard time. I really appreciate you helping me out!

**Comment by balzack** [comment, marked as answer]
No, I believe this is my fault. Recently I added MFA support which requires the latest webapp. I needed to update the documentation; this is the the latest webapp:
https://github.com/balzack/databag/releases/download/v1.1.1019/webapp.tar.gz

I've tested this build as part of the container, but I will do additional testing on your similar setup. I'm sorry for your trouble. The above link should resolve the problem.

**Reply by reluss** [reply]
Now it works! Thank you very much!

**Reply by balzack** [reply]
great! ....thank you for your patience and making me aware of the problem

**Reply by balzack** [reply]
BTW, there are some limitations using 'http' instead of 'https'. Federating with other nodes won't work, and the fdroid mobile app disables non 'https' communication. You also might see that 'copy to clipboard' not working, which is a browser limitation when not using 'https'

**Reply by reluss** [reply]
Yeah I just noticed. Since I haven't done anything like that before I now need to find a good tutorial on how to set it up. Maybe you know a good source for that, that can be of help for others too.

**Reply by balzack** [reply]
It depends on your preference:
- for getting a dns name, you can buy a domain name or use a free service like [this](https://www.noip.com/)
- for getting a certificate you can install certbot directly or you can use a separate reverse proxy (which likely uses certbot internally)
- some people like to use a cloudflare tunnel which will hide their public IP which will also take care of configuring your domain name and proxy

If you choose one, I can help you along the process or look up resources for you. I will be AFK for the rest of the day though.
